### DISCUSSION TITLE
OIDC authentification with Discord does not work

### DISCUSSION THREAD

**Discussion by unordentlich** [outline/outline#3602] [Self Hosting] [answered]
Hey! 

I wanted to set up an OIDC authentification for my outline instance with discord and it returned the error parameter `?notice=authentication-required` as well as this error log in the docker console: https://www.toptal.com/developers/hastebin/jehokoyiqe.rb 

How can I fix this? Is there a way to set up OIDC login without providing an email address?
Thank you very much in advance! :)

**Comment by Limezy** [comment, marked as answer]
I'm afraid you'll have to "monkey-patch" the OIDC passport.js strategy to get it working.
I guess you could try to create a fake email using the user ID or name on this line : 
https://github.com/outline/outline/blob/da87fd422d3e6cfa8da1caed543538449890ea80/server/routes/auth/providers/oidc.ts#L91
This will obviously have some bad consequences, starting with a non-working onboarding email for new users... 

You could also do a cleaner job using the dedicated passport.js strategy and create a new auth provider
https://www.passportjs.org/packages/passport-discord/

In both cases, you'll have to patch and compile from sources.
In any case, should you get that second cleaner option working, I'm not sure that @tommoor would accept a PR (see here : https://github.com/outline/outline/pull/2470#issuecomment-982259704)

**Reply by unordentlich** [reply]
Thank you very much! 😄
